src/OrbitCore/Path.cpp

```diff
@@ -92,14 +97,36 @@ std::filesystem::path CreateOrGetOrbitAppDataDir() {
   return path;
 }
 
-std::filesystem::path CreateOrGetOrbitUserDataDir() {
+static std::filesystem::path GetDocumentsPath() {
 #ifdef WIN32
-  std::filesystem::path path = std::filesystem::path(GetEnvVar("USERPROFILE"));
+  PWSTR ppszPath;
+  HRESULT result = SHGetKnownFolderPath(FOLDERID_Documents, 0, nullptr, &ppszPath);
+
+  if (result != S_OK) {
+    CoTaskMemFree(ppszPath);
+    LPSTR error_string = nullptr;
+    FormatMessageA(
+        FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+        nullptr, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+        reinterpret_cast<LPSTR>(&error_string), 0, nullptr);
+    std::filesystem::path path = std::filesystem::path(GetEnvVar("USERPROFILE")) / "Documents";
+    ERROR("Retrieving path to Documents (defaulting to \"%s\"): %s", path.string(), error_string);
+    LocalFree(error_string);
+    return path;
+  }
+
+  std::wstring wpath = ppszPath;
+  CoTaskMemFree(ppszPath);
+  std::string path{wpath.begin(), wpath.end()};
```

Comment:
This conversion could be problematic if the path contains non-ascii-characters. What about creating a `std::filesystem::path` directory from `wpath`. There is actually a constructor for that purpose.

Reply:
Oh good point, done. (`LOG` still needs `string` it seems)